src/de/darmstadt/tu/crossing/DigestInputStream.cryptsl

```diff
@@ -0,0 +1,33 @@
+SPEC java.security.DigestInputStream
+OBJECTS
+    boolean on;
+    byte[] b;
+    int off;
+    int len;
+    java.io.InputStream is;
+    java.security.MessageDigest md;
+
+EVENTS
+    c1: DigestInputStream(is, md);
+
+    g: md = getMessageDigest();
+
+    o: on(on);
+
+    r1: read();
+    r2: read(b, off, len);
+    Reads := r1 | r2;
+
+    c: close();
+
+FORBIDDEN
+    o;
+
+ORDER
+    c1, g*, Reads, g, c
```

Comment:
Shouldn't it just be c1, Reads, c

The getMessageDigest is fully optional:
`MessageDigest m = MessasgeDigest.getInstanec("...")`
`DigestInputStream d = DigestInputStream(is,m)`
`//I have access to m anyway`

Reply:
I agree with Johannes. In fact, remove the getMessageDigest method from the rule entirely as we do not need it.

Reply:
Yeah, I also had a doubt on whether it was needed or not. I will remove it now and update the ORDER section